hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/transfer/AssociateTokenRecipientsStep.java

```diff
@@ -168,10 +168,12 @@ private TokenAssociation validateAndBuildAutoAssociation(
         final var tokenRel = tokenRelStore.get(accountId, tokenId);
         final var config = handleContext.configuration();
 
-        if (tokenRel == null && account.maxAutoAssociations() > 0) {
-            validateFalse(
-                    account.usedAutoAssociations() >= account.maxAutoAssociations(),
-                    NO_REMAINING_AUTOMATIC_ASSOCIATIONS);
+        if (tokenRel == null && (account.maxAutoAssociations() > 0 || account.maxAutoAssociations() == -1)) {
```

Comment:
This could probably be simplified

Reply:
This collapses to the following if you simplify with the understanding that the value cannot be `< -1`:
```suggestion
        if (tokenRel == null && (account.maxAutoAssociations() != 0)) {
```

Reply:
I applied the above suggestion.